integration-test/src/main/java/com/sequenceiq/it/cloudbreak/actor/CloudbreakUserCache.java

```diff
@@ -1,66 +1,158 @@
+        this.usersByAccount = users;
+    }
+
+    public Map<String, List<CloudbreakUser>> getUsersByAccount() {
+        return usersByAccount;
+    }
+
+    public String getRealUmsUserAccount() {
+        return realUmsUserAccount;
     }
 
-    public void initUsers() {
+    public void setRealUmsUserAccount(String realUmsUserAccountKey) {
+        this.realUmsUserAccount = realUmsUserAccountKey;
+    }
+
+    public String getRealUmsUserEnvironment() {
+        return realUmsUserEnvironment;
+    }
+
+    public void setRealUmsUserEnvironment(String realUmsUserEnvironmentKey) {
+        this.realUmsUserEnvironment = realUmsUserEnvironmentKey;
+    }
+
+    public void setInitializedKeys(Map<String, String> keys) {
+        this.initializedKeys = keys;
+    }
+
+    public void initUsers(String environmentKey, String accountKey) {
         String userConfigPath = "ums-users/api-credentials.json";
+        LOGGER.info("Real UMS users are initializing by environment: {} and account: {}. Repository is present at: {} path", environmentKey, accountKey,
+                userConfigPath);
         try {
-            this.usersByAccount = JsonUtil.readValue(
-                    FileReaderUtils.readFileFromClasspathQuietly(userConfigPath), new TypeReference<>() {
-                    });
-        } catch (IOException e) {
-            throw new RuntimeException(String.format("Can't read file: %s It's possible you did run make fetch-secrets", userConfigPath));
+            String accountId = null;
+            List<CloudbreakUser> cloudbreakUsers = new ArrayList<CloudbreakUser>();
+            JSONObject usersByEnvAndAcc = new JSONObject(FileReaderUtils.readFileFromClasspathQuietly(userConfigPath));
+            JSONArray devEnvironment = usersByEnvAndAcc.getJSONArray(environmentKey);
+            for (int i = 0; i < devEnvironment.length(); i++) {
+                JSONObject jsonObject1 = (JSONObject) devEnvironment.get(i);
+                JSONArray jsonarray1 = (JSONArray) jsonObject1.get(accountKey);
+                for (int j = 0; j < jsonarray1.length(); j++) {
+                    accountId = Crn.fromString(((JSONObject) jsonarray1.get(j)).getString("crn")).getAccountId();
+                    String displayName = ((JSONObject) jsonarray1.get(j)).getString("displayName");
+                    String desc = ((JSONObject) jsonarray1.get(j)).getString("desc");
+                    String crn = ((JSONObject) jsonarray1.get(j)).getString("crn");
+                    String accessKey = ((JSONObject) jsonarray1.get(j)).getString("accessKey");
+                    String secretKey = ((JSONObject) jsonarray1.get(j)).getString("secretKey");
+                    boolean admin = Boolean.parseBoolean(((JSONObject) jsonarray1.get(j)).getString("admin"));
+                    cloudbreakUsers.add(new CloudbreakUser(accessKey, secretKey, displayName, crn, desc, admin));
+                }
```

Comment:
to be honest, i am not a huge fan of extracting fields from jsonarray manually, we should create a model which has same structure and parse it, like before